tools/themis/src/rules.rs

```diff
@@ -0,0 +1,329 @@
+        let (rust_version, raw) = match pkg.raw["package"].get("rust-version") {
+            Some(rust_version) => {
+                let raw = rust_version.as_str().unwrap();
+                (semver::VersionReq::parse(raw)?, raw)
+            }
+            // we can skip, since we have has_rust_version check
+            None => continue,
+        };
+
+        if !rust_version.matches(&rust_toolchain) {
+            outliers.push(PackageOutcome { pkg, value: Some(raw.to_owned()) });
+        }
+    }
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages have an incompatible `rust-version`".to_string(),
+            expected: Some(Expected {
+                value: format!("<={}", rust_toolchain),
+                reason: Some("as defined in the `rust-toolchain`".to_string()),
+            }),
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+/// ensure all crates share the same rust edition
+pub fn has_unified_rust_edition(workspace: &Workspace) -> Result<(), Error> {
+    let mut edition_groups = HashMap::new();
+
+    for pkg in &workspace.members {
+        *edition_groups.entry(&pkg.parsed.edition).or_insert(0) += 1;
+    }
+
+    let (most_common_edition, n_compliant) =
+        edition_groups.into_iter().reduce(|a, b| if a.1 > b.1 { a } else { b }).unwrap();
+
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| pkg.parsed.edition != *most_common_edition)
+        .map(|pkg| PackageOutcome { pkg, value: Some(pkg.parsed.edition.clone()) })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages have an unexpected rust edition".to_string(),
+            expected: Some(Expected {
+                value: most_common_edition.clone(),
+                reason: Some(format!("used by {} other packages in the workspace", n_compliant)),
+            }),
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+const EXPECTED_AUTHOR: &str = "Near Inc <hello@nearprotocol.com>";
+
+/// ensure all crates have the appropriate author, non-exclusively of course.
+pub fn author_is_near(workspace: &Workspace) -> Result<(), Error> {
+    let outliers = workspace
+        .members
+        .iter()
+        .filter(|pkg| !pkg.parsed.authors.iter().any(|author| author == EXPECTED_AUTHOR))
+        .map(|pkg| PackageOutcome { pkg, value: Some(format!("{:?}", pkg.parsed.authors)) })
+        .collect::<Vec<_>>();
+
+    if !outliers.is_empty() {
+        return Err(Error::OutcomeError {
+            msg: "These packages need to be correctly authored".to_string(),
+            expected: Some(Expected { value: EXPECTED_AUTHOR.to_string(), reason: None }),
+            outliers,
+        });
+    }
+
+    Ok(())
+}
+
+/// ensure all non-private crates have a license
+pub fn publishable_has_license(workspace: &Workspace) -> Result<(), Error> {
```

Comment:
Rethinking this, maybe this check should be for all crates, not just non-private ones.